spring-boot/admin/src/main/java/eu/coatrack/admin/service/OauthUserAccountManagement.java

```diff
@@ -0,0 +1,119 @@
+ * #%L
+ * coatrack-admin
+ * %%
+ * Copyright (C) 2013 - 2022 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import eu.coatrack.config.github.GithubEmail;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+public class OauthUserAccountManagement {
+
+    private static final String GITHUB_API_EMAIL = "https://api.github.com/user/emails";
+
+    @Autowired
+    private OAuth2AuthorizedClientService clientService;
+
+    private OAuth2User getLoggedInUser() {
+        return (OAuth2User) SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getPrincipal();
+    }
+
+    public String getLoginNameFromLoggedInUser() {
+        return getLoggedInUser().getAttribute("login");
+    }
+
+    public String getNameFromLoggedInUser() {
+        return getLoggedInUser().getAttribute("name");
+    }
+
+    public String getCompanyFromLoggedInUser() {
+        return getLoggedInUser().getAttribute("company");
+    }
+
+    public String getEmailFromLoggedInUser() throws JsonProcessingException {
+        String email = getLoggedInUser().getAttribute("email");
+
+        if (wasGitHubMailAddressSetToPrivateAndThereforeNotDirectlyReadable(email)) {
+            ResponseEntity<String> emailListFromGithub = getEmailsListFromGithub();
+            email = getPrimaryEmailFromLoggedInUser(emailListFromGithub);
```

Comment:
why not encapsulate that "email-list stuff" in the method you call here? I would suggest:

```suggestion
            email = getPrimaryGithubEmailForLoggedInUser();
```
And handle that email list stuff inside that method.